src/editor.py

```diff
@@ -1198,33 +1203,31 @@ def handle_input(self):
 					if val == "d":
 						self.selectedNote = self.create_note(
 							note["beatpos"][0]*4 + note["beatpos"][1], 
-							note["key"]
+							note.get("key")
 						)
 						self.mapToEdit["notes"][self.selectedNote] = copy.deepcopy(note)
 
-					if note["type"] == "hit_object":
+					if note["type"] == "hit_object" and val:
 						screenPos = note["screenpos"]
 						calculatedPos = Game.calculatePosition(screenPos, 5, 3, term.width-10, term.height-11)
-						if val == "h":
+						# erase hitbox
+						if val in "hjklHJKL":
 							print_at(calculatedPos[0]-1, calculatedPos[1]-1, f"{term.normal}   ")
 							print_at(calculatedPos[0]-1, calculatedPos[1]+0, f"{term.normal}   ")
 							print_at(calculatedPos[0]-1, calculatedPos[1]+1, f"{term.normal}   ")
-							note["screenpos"][0] = max(round(note["screenpos"][0] - 1/(defaultSize[0]-1), 3), 0)
-						if val == "j":
-							print_at(calculatedPos[0]-1, calculatedPos[1]-1, f"{term.normal}   ")
-							print_at(calculatedPos[0]-1, calculatedPos[1]+0, f"{term.normal}   ")
-							print_at(calculatedPos[0]-1, calculatedPos[1]+1, f"{term.normal}   ")
-							note["screenpos"][1] = min(round(note["screenpos"][1] + 1/defaultSize[1], 3), 1)
-						if val == "k":
-							print_at(calculatedPos[0]-1, calculatedPos[1]-1, f"{term.normal}   ")
-							print_at(calculatedPos[0]-1, calculatedPos[1]+0, f"{term.normal}   ")
-							print_at(calculatedPos[0]-1, calculatedPos[1]+1, f"{term.normal}   ")
-							note["screenpos"][1] = max(round(note["screenpos"][1] - 1/defaultSize[1], 3), 0)
-						if val == "l":
-							print_at(calculatedPos[0]-1, calculatedPos[1]-1, f"{term.normal}   ")
-							print_at(calculatedPos[0]-1, calculatedPos[1]+0, f"{term.normal}   ")
-							print_at(calculatedPos[0]-1, calculatedPos[1]+1, f"{term.normal}   ")
-							note["screenpos"][0] = min(round(note["screenpos"][0] + 1/(defaultSize[0]-1), 3), 1)
+						# calculate new x/y position
```

Comment:
yeah previously that section was kinda badly written, thanks for rewriting it in a clearer way-